### ISSUE TITLE
[Bruno-cli] HTML Report fails to generate if API response contains JS replace sequences

### ISSUE THREAD

**Issue by Iilun** [usebruno/bruno#3311]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

When an API response contains one of the [replace special commands](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_string_as_the_replacement), the HTML report can generate invalid HTML.

In my specific case, the API answers with a string containing "$'", so the HTML reports duplicates the end part multiple times.

To reproduce, use the below bru file with an API answering with a "$'" anywhere.

Then use 
```
bru run  --reporter-html results.html
```

The report is then invalid HTML

A simple fix would be, in the `reporters/html.js` file, to replace all $ in the results with $$ to prevent these sequences (this could have side effects)

```
const resultsJson = JSON.stringify(results, null, 2).replaceAll("$","$$$$");
```

### .bru file to reproduce the bug

```
meta {
  name: Request a response
  type: http
  seq: 1
}

get {
  url: https://-an url sending the below body as a response-
  body: json
  auth: none
}

body:json {
  {
  "detail": "i will destroy the html report - regex is '^reportdestroyer$'"
  }
}
```

### Screenshots/Live demo link

The request

<img width="1210" alt="image" src="https://github.com/user-attachments/assets/0ad6cc61-d51c-47a7-a816-5a050100f31d">

The report

<img width="1910" alt="image" src="https://github.com/user-attachments/assets/8656eddb-1e95-4417-91b0-fb51313d25a1">

The [HTML file (with bogus type so gh allows it)](https://github.com/user-attachments/files/17377560/results.patch) is here, we can see line 402 that the replace sequence has been replaced with HTML code

**Comment by helloanoop** [comment]
@anusreesubash is currently looking into this, and we should be able to provide an update within the next week.

**Comment by sreelakshmi-bruno** [comment]
Hi @Iilun, this issue has been addressed in our latest release( 1.34.1). Please check it out. Thanks!
